From 2bb7d74215bf6ae4f350de11bff04b68cf3c82b3 Mon Sep 17 00:00:00 2001
From: Bruno Friedmann <bruno.friedmann@bareos.com>
Date: Mon, 27 Mar 2023 13:22:11 +0200
Subject: [PATCH] core: fixes for gcc-13 support

- add explicit system_error and stdexcept
- console_conf.cc fix std::move warning

Signed-off-by: Bruno Friedmann <bruno.friedmann@bareos.com>
---
 core/src/cats/cats.h                    | 2 ++
 core/src/console/console_conf.cc        | 4 ++--
 core/src/lib/thread_list.cc             | 3 ++-
 core/src/lib/thread_specific_data.cc    | 3 ++-
 core/src/lib/thread_specific_data.h     | 3 ++-
 core/src/lib/thread_specific_data_key.h | 4 +++-
 6 files changed, 13 insertions(+), 6 deletions(-)

diff --git core/src/cats/cats.h.orig core/src/cats/cats.h
index 7a6641b3f72..edb5ecbe38b 100644
--- core/src/cats/cats.h.orig
+++ core/src/cats/cats.h
@@ -41,6 +41,8 @@
 #include "lib/output_formatter.h"
 
 #include <string>
+#include <stdexcept>
+#include <system_error>
 #include <vector>
 template <typename T> class dlist;
 
diff --git core/src/console/console_conf.cc.orig core/src/console/console_conf.cc
index 9d3aecead2b..f0c0402744e 100644
--- core/src/console/console_conf.cc.orig
+++ core/src/console/console_conf.cc
@@ -3,7 +3,7 @@
 
    Copyright (C) 2000-2009 Free Software Foundation Europe e.V.
    Copyright (C) 2011-2012 Planets Communications B.V.
-   Copyright (C) 2013-2022 Bareos GmbH & Co. KG
+   Copyright (C) 2013-2023 Bareos GmbH & Co. KG
 
    This program is Free Software; you can redistribute it and/or
    modify it under the terms of version three of the GNU Affero General Public
@@ -203,7 +203,7 @@ static bool SaveResource(int type, ResourceItem* items, int pass)
                 res_dir->resource_name_);
         } else {
           p->tls_cert_.allowed_certificate_common_names_
-              = std::move(p->tls_cert_.allowed_certificate_common_names_);
+              = std::move(res_dir->tls_cert_.allowed_certificate_common_names_);
         }
         break;
       }
diff --git core/src/lib/thread_list.cc.orig core/src/lib/thread_list.cc
index 50aefb8ee2d..05607124a55 100644
--- core/src/lib/thread_list.cc.orig
+++ core/src/lib/thread_list.cc
@@ -1,7 +1,7 @@
 /*
    BAREOS® - Backup Archiving REcovery Open Sourced
 
-   Copyright (C) 2019-2022 Bareos GmbH & Co. KG
+   Copyright (C) 2019-2023 Bareos GmbH & Co. KG
 
    This program is Free Software; you can redistribute it and/or
    modify it under the terms of version three of the GNU Affero General Public
@@ -32,6 +32,7 @@
 #include <memory>
 #include <mutex>
 #include <set>
+#include <system_error>
 #include <thread>
 
 static constexpr int debuglevel{800};
diff --git core/src/lib/thread_specific_data.cc.orig core/src/lib/thread_specific_data.cc
index 0953f5c125a..22a98c4a759 100644
--- core/src/lib/thread_specific_data.cc.orig
+++ core/src/lib/thread_specific_data.cc
@@ -3,7 +3,7 @@
 
    Copyright (C) 2000-2012 Free Software Foundation Europe e.V.
    Copyright (C) 2011-2012 Planets Communications B.V.
-   Copyright (C) 2019-2019 Bareos GmbH & Co. KG
+   Copyright (C) 2019-2023 Bareos GmbH & Co. KG
 
    This program is Free Software; you can redistribute it and/or
    modify it under the terms of version three of the GNU Affero General Public
@@ -25,6 +25,7 @@
 #include "lib/thread_specific_data.h"
 #include "lib/thread_specific_data_key.h"
 #include "include/jcr.h"
+#include <system_error>
 
 JobControlRecord* GetJcrFromThreadSpecificData()
 {
diff --git core/src/lib/thread_specific_data.h.orig core/src/lib/thread_specific_data.h
index 4bd4dedafae..4c427d171c0 100644
--- core/src/lib/thread_specific_data.h.orig
+++ core/src/lib/thread_specific_data.h
@@ -3,7 +3,7 @@
 
    Copyright (C) 2000-2012 Free Software Foundation Europe e.V.
    Copyright (C) 2011-2012 Planets Communications B.V.
-   Copyright (C) 2019-2021 Bareos GmbH & Co. KG
+   Copyright (C) 2019-2023 Bareos GmbH & Co. KG
 
    This program is Free Software; you can redistribute it and/or
    modify it under the terms of version three of the GNU Affero General Public
@@ -24,6 +24,7 @@
 #ifndef BAREOS_LIB_THREAD_SPECIFIC_DATA_H_
 #define BAREOS_LIB_THREAD_SPECIFIC_DATA_H_
 
+
 class JobControlRecord;
 
 JobControlRecord* GetJcrFromThreadSpecificData();
diff --git core/src/lib/thread_specific_data_key.h.orig core/src/lib/thread_specific_data_key.h
index 9c02c07f47e..c8c63c757fe 100644
--- core/src/lib/thread_specific_data_key.h.orig
+++ core/src/lib/thread_specific_data_key.h
@@ -1,7 +1,7 @@
 /*
    BAREOS® - Backup Archiving REcovery Open Sourced
 
-   Copyright (C) 2019-2021 Bareos GmbH & Co. KG
+   Copyright (C) 2019-2023 Bareos GmbH & Co. KG
 
    This program is Free Software; you can redistribute it and/or
    modify it under the terms of version three of the GNU Affero General Public
@@ -22,6 +22,8 @@
 #ifndef BAREOS_LIB_THREAD_SPECIFIC_DATA_KEY_H_
 #define BAREOS_LIB_THREAD_SPECIFIC_DATA_KEY_H_
 
+
+#include <system_error>
 #include "lib/berrno.h"
 
 #include <mutex>
